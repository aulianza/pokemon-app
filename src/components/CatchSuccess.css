.catch-success {
    text-align: -webkit-center;
    margin-top: 20px;
}

.error-message {
    color: #f44336;
    padding-top: 20px;
}

form {
    width: 70%;
}

form > label {
    font-size: 1.1em;
}

form > input {
    display: block;
    width: 98%;
    height: 2rem;
    font-size: 1.1rem;
    text-align: center;
    margin: 15px 0;
}