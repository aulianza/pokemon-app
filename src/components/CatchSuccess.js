import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import getOwnedPokemon from "../graphql/queries/getOwnedPokemon";
import addOwnedPokemon from "../graphql/operations/addOwnedPokemon";
import Button from "../components/Button";
import styled from "@emotion/styled";

const CatchSuccess = (props) => {
    const [statusState, setStatusState] = useState(false);
    let route = useHistory();

    const addToMyPokemon = (e) => {
        e.preventDefault();

        const nickname = e.target.nickname.value;
        const newPokemonData = {
            id: props.pokeId,
            nickname: nickname,
            name: props.pokeName,
            image: props.pokeImage,
        };

        if (checkNickname(nickname) && nickname) {
            addOwnedPokemon(newPokemonData);
            setStatusState(false);
            route.push("/my-pokemon");
        } else {
            setStatusState(true);
        }
    };

    const checkNickname = (nickname) => {
        const myOwnedPokemon = getOwnedPokemon();
        const check = myOwnedPokemon.filter((pokemon) => {
            return pokemon.nickname === nickname;
        });

        if (check.length) {
            return false;
        } else {
            return true;
        }
    };

    return (
        <>
            <img
                src={props.pokeImage}
                alt="Pokemon Detail"
                width="140"
                height="140"
            />
            <StyledCatchSuccess>
                <form onSubmit={addToMyPokemon}>
                    <label>Give Nickname :</label>
                    <input
                        type="text"
                        name="nickname"
                        placeholder="Input here..."
                        autoFocus
                    ></input>
                    <Button
                        type="submit"
                        buttonColor="btn-success"
                        title="Add to My Pokemon"
                        onButtonClicked={() => console.log("success")}
                        isBlock
                    />
                </form>
                {statusState ? (
                    <div className="error-message">
                        <code className="red-text">nickname exist or the field empty</code>
                    </div>
                ) : (
                    <span></span>
                )}
            </StyledCatchSuccess>
        </>
    );
};

export default CatchSuccess;

const StyledCatchSuccess = styled.div`
    text-align: -webkit-center;
    margin-top: 20px;

    .error-message {
        color: #f44336;
        padding-top: 20px;
    }

    form {
        width: 70%;
    }

    form > label {
        font-size: 1.1em;
    }

    form > input {
        display: block;
        width: 98%;
        height: 2rem;
        font-size: 1.1rem;
        text-align: center;
        margin: 15px 0;
    }
`;
