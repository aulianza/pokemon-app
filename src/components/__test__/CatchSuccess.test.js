import CatchSuccess from "../CatchSuccess";
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from "react-dom/test-utils";

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("loading-component, snapshot", () => {
    act(() => {
        render(<CatchSuccess />, container)
    })
    expect(container).toMatchSnapshot();
});
