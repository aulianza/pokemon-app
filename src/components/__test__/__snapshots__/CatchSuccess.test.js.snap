// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading-component, snapshot 1`] = `
<div>
  <img
    alt="Pokemon Detail"
    height="140"
    width="140"
  />
  <div
    class="css-ew1di0"
  >
    <form>
      <label>
        Give Nickname :
      </label>
      <input
        name="nickname"
        placeholder="Input here..."
        type="text"
      />
      <button
        class="css-l6cmt2"
      >
        Add to My Pokemon
      </button>
    </form>
    <span />
  </div>
</div>
`;
